src/foam/java/refinements.js

```diff
@@ -321,6 +321,33 @@ foam.CLASS({
   ]
 });
 
+foam.CLASS({
+  refines: 'foam.core.Action',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'javaCode'
+    },
+    {
+      class: 'String',
+      name: 'javaReturns'
```

Comment:
For JS, Actions (like most implementations of the Command Design Pattern) don't return values.